### DISCUSSION TITLE
executing commands in the background

### DISCUSSION THREAD

**Discussion by imed-ghomari** [Taitava/obsidian-shellcommands#188] [Ideas - on hold]
I set up a command with the event `every 30 minutes`. I keep obsidian open and hidden using this [plugin's](https://github.com/mjessome/obsidian-global-hotkeys) `hide/show obsidian` command. 

The problem is that every 30 minutes, when it's time to execute the command, obsidian is back in the foreground. Is there a way to keep obsidian hidden while doing this ?

I use bash (mac).

**Comment by Taitava** [comment]
Sorry, I don't know how to prevent Obsidian from coming back to foreground.

SC uses Node.js's [child_process](https://nodejs.org/api/child_process.html) to execute shell commands. I tried to make a quick search on Google to find out if someone else has had a situation where `child_process` would bring an application window back to foreground, but I didn't find anything.

I cannot even test if it's specifically the shell command execution part that brings Obsidian back to foreground, or something else that causes it. E.g. could JavaScript's [setInterval()](https://www.w3schools.com/Jsref/met_win_setinterval.asp) cause it? That's what triggers the _Every n seconds_ event. I guess not, but I'm not sure.

I tried to test hiding Obsidian using the plugin you mentioned, but it didn't hide Obsidian on my Xubuntu (Linux) machine nor on my Windows machine. Obsidian just goes behind all other windows, but is not _hidden_.

If someone comes up with some tips on how to prevent Obsidian from coming back to foreground, I might be able to do something, but currently I unfortunately do not have a solution for this, so I'll move this to _Ideas - on hold_ for now.

**Reply by imed-ghomari** [reply]
It was advanced uri plugin that kept bringing obsidian to foreground not SC. [source](https://discord.com/channels/686053708261228577/707816848615407697/958051434614632459)

**Reply by Taitava** [reply]
Ok, thanks for the info! 🙂
